Modules/_datetimemodule.c

```diff
@@ -1950,11 +1955,20 @@ wrap_strftime(PyObject *object, PyObject *format, PyObject *timetuple,
             ntoappend = PyBytes_GET_SIZE(freplacement);
         }
 #ifdef Py_NORMALIZE_CENTURY
-        else if (ch == 'Y' || ch == 'G') {
+        else if (ch == 'Y' || ch == 'G'
+#ifdef Py_STRFTIME_C99_SUPPORT
+                 || ch == 'F' || ch == 'C'
+#endif
+        ) {
             /* 0-pad year with century as necessary */
-            PyObject *item = PyTuple_GET_ITEM(timetuple, 0);
+            if (!PyTuple_Check(timetuple)) {
+                goto Done;
```

Comment:
This returns NULL without raising an exception.

There are several ways to write this code correctly:

* Use `PyTuple_Check()`, and if it returns false, set an error and return. If it returns true, you can use `PyTuple_GetItem()`.
* Use `PyTuple_Check()` and `PyTuple_GET_SIZE()` to check for non-empty tuple. If the check is failed, set an error and return. If it is passed, you can use `PyTuple_GET_ITEM()`.
* Use `PySequence_GetItem()`. But note that it returns a new reference, you should decref it after calling `PyLong_AsLong()`.

All ways have almost the same complexity and performance, and the differences in behavior here are insignificant. Therefore, I leave the choice to you.

Reply:
Ah you're absolutely right. I love the idea of switching to `PySequence_GetItem` instead to maximize code reuse. Thanks so much for offering all the options.